src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -198,6 +229,89 @@ public void cancelScheduledOperations(Long programmingExerciseId) {
+     */
+    public void validateNewProgrammingExerciseSettings(ProgrammingExercise programmingExercise, Course course) {
+        if (programmingExercise.getId() != null) {
+            throw new BadRequestAlertException("A new programmingExercise cannot already have an ID", "Exercise", "idexists");
+        }
+
+        programmingExercise.validateGeneralSettings();
+        programmingExercise.validateProgrammingSettings();
+        programmingExercise.validateManualFeedbackSettings();
+        auxiliaryRepositoryService.validateAndAddAuxiliaryRepositoriesOfProgrammingExercise(programmingExercise, programmingExercise.getAuxiliaryRepositories());
+        submissionPolicyService.validateSubmissionPolicyCreation(programmingExercise);
+
+        ProgrammingLanguageFeature programmingLanguageFeature = programmingLanguageFeatureService.get()
+                .getProgrammingLanguageFeatures(programmingExercise.getProgrammingLanguage());
+
+        validatePackageName(programmingExercise, programmingLanguageFeature);
+        validateProjectType(programmingExercise, programmingLanguageFeature);
+
+        // Check if checkout solution repository is enabled
+        if (programmingExercise.getCheckoutSolutionRepository() && !programmingLanguageFeature.isCheckoutSolutionRepositoryAllowed()) {
+            throw new BadRequestAlertException("Checkout solution repository is not supported for this programming language", "Exercise", "checkoutSolutionRepositoryNotSupported");
+        }
+
+        programmingExerciseRepository.validateCourseSettings(programmingExercise, course);
+        validateStaticCodeAnalysisSettings(programmingExercise);
+
+        programmingExercise.generateAndSetProjectKey();
+        checkIfProjectExists(programmingExercise);
+
+    }
+
+    private void validateProjectType(ProgrammingExercise programmingExercise, ProgrammingLanguageFeature programmingLanguageFeature) {
+        // Check if project type is selected
+        if (!programmingLanguageFeature.getProjectTypes().isEmpty()) {
+            if (programmingExercise.getProjectType() == null) {
+                throw new BadRequestAlertException("The project type is not set", "Exercise", "projectTypeNotSet");
+            }
+            if (!programmingLanguageFeature.getProjectTypes().contains(programmingExercise.getProjectType())) {
+                throw new BadRequestAlertException("The project type is not supported for this programming language", "Exercise", "projectTypeNotSupported");
+            }
+        }
+        else if (programmingExercise.getProjectType() != null) {
+            throw new BadRequestAlertException("The project type is set but not supported", "Exercise", "projectTypeSet");
+
+        }
+    }
+
+    private void validatePackageName(ProgrammingExercise programmingExercise, ProgrammingLanguageFeature programmingLanguageFeature) {
+        // Check if package name is set
+        if (programmingLanguageFeature.isPackageNameRequired()) {
```

Comment:
```suggestion
        if (!programmingLanguageFeature.isPackageNameRequired()) {
            return;
        }
```
Optional: inverting the condition would avoid additional nesting